code/datums/chatmessage.dm

```diff
@@ -152,16 +161,18 @@
 	owned_by.images |= message
 	animate(message, alpha = 255, time = CHAT_MESSAGE_SPAWN_TIME)
 
-	// Prepare for destruction
+	// Register with the runechat SS to handle EOL and destruction
 	scheduled_destruction = world.time + (lifespan - CHAT_MESSAGE_EOL_FADE)
-	addtimer(CALLBACK(src, .proc/end_of_life), lifespan - CHAT_MESSAGE_EOL_FADE, TIMER_UNIQUE|TIMER_OVERRIDE)
+	enter_subsystem()
 
 /**
-  * Applies final animations to overlay CHAT_MESSAGE_EOL_FADE deciseconds prior to message deletion
+  * Applies final animations to overlay CHAT_MESSAGE_EOL_FADE deciseconds prior to message deletion,
+  * sets time for scheduling deletion and re-enters the runechat SS for qdeling
   */
 /datum/chatmessage/proc/end_of_life(fadetime = CHAT_MESSAGE_EOL_FADE)
+	eol_complete = scheduled_destruction + CHAT_MESSAGE_EOL_FADE
```

Comment:
Any reason you didn't use `fadetime` here?

`eol_complete = scheduled_destruction + fadetime`

Reply:
there is no reason, that should be fadetime you are correct